adagrams/game.py

```diff
@@ -1,11 +1,134 @@
+            else:
+                #...and if character in word matches said letter...
+                if char == key:
+                    #...then add corresponding point value to word's score
+                    score += score_chart[key]
+                    
+    #if the word is greater than 6 characters...
+    if len(word) > 6:
+        #...then add additional 8 pts to word's score
+        score += 8
+    
+    #return total score for word
+    return score   
+
+                
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #Goal: returns tuple containing winning word and highest score
+    # in the case of tie in scores, use these tie-breaking rules:
+    # prefer the word with the fewest letters...
+    # ...unless one word has 10 letters. If the top score is tied between multiple words and one is 10 letters long, choose the one with 10 letters over the one with fewer tiles
+    # If there are multiple words that are the same score and the same length, pick the first one in the supplied list
+    
+    #initialize winning_words_list list and word_score_dict dictionary
+    winning_words_list = []
+    word_score_dict = {}
+    #loop thru each word in list of words
+    for word in word_list:
+        #invoke function score_word to calculate score of each word looped thru
+        score = score_word(word)
+        #populate word_score_dict dictionary with word:score pairs
+        word_score_dict[word] = score
+    #get the word or words with the highest score from the dict and store in string_with_highest_score variable
+    string_with_highest_score = max(word_score_dict, key=word_score_dict.get)
+    #get the highest score from the dict
+    highest_score = word_score_dict[string_with_highest_score]
+    #print(highest_score) #verify
+    #loop thru dict
+    for key, value in word_score_dict.items():
+        #if score is equal to the highest score, then add word from dict to winning_words_list to get list of highest-scoring words
+        if value == highest_score:
+            winning_words_list.append(key)
+
+    #print(winning_words_list) #verify
+    
+    #if there's only 1 highest-scoring word, then return such word w/ score
+    if len(winning_words_list) == 1:
+        #print(winning_words_list[0])
+        return winning_words_list[0], score_word(winning_words_list[0])
+    #if there're > 1 highest-scoring words, then...
+    else:
+        #...look thru list of highest-scoring words, find the first word w/ 10 letters
+        winning_word_with_ten_letters = next((word for word in winning_words_list if len(word) == 10), None)
+        #if there's such a 10-letter word, then return it w/ score
+        if winning_word_with_ten_letters != None:
+            #print(winning_word_with_ten_letters) #verify
+            return winning_word_with_ten_letters, score_word(winning_word_with_ten_letters)
```

Comment:
[nit] You already calculated the score, so you can reuse it

```suggestion
            return winning_word_with_ten_letters, word_score_dict[winning_word_with_ten_letters]
```